vault/logical_system_raft.go

```diff
@@ -335,6 +382,112 @@ func (b *SystemBackend) handleStorageRaftSnapshotRead() framework.OperationFunc
+				"failure_tolerance":            health.FailureTolerance,
+				"optimistic_failure_tolerance": health.OptimisticFailureTolerance,
+				"servers":                      health.Servers,
+				"leader":                       health.Leader,
+				"voters":                       health.Voters,
+				"read_replicas":                health.ReadReplicas,
+			},
+		}, nil
+	}
+}
+
+func (b *SystemBackend) handleStorageRaftAutopilotConfigRead() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		raftStorage, ok := b.Core.underlyingPhysical.(*raft.RaftBackend)
+		if !ok {
+			return logical.ErrorResponse("raft storage is not in use"), logical.ErrInvalidRequest
+		}
+		config := raftStorage.AutopilotConfig()
+		return &logical.Response{
+			Data: map[string]interface{}{
+				"cleanup_dead_servers":      config.CleanupDeadServers,
+				"last_contact_threshold":    config.LastContactThreshold.String(),
+				"max_trailing_logs":         config.MaxTrailingLogs,
+				"min_quorum":                config.MinQuorum,
+				"server_stabilization_time": config.ServerStabilizationTime.String(),
+			},
+		}, nil
+	}
+}
+
+func (b *SystemBackend) handleStorageRaftAutopilotConfigUpdate() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		raftStorage, ok := b.Core.underlyingPhysical.(*raft.RaftBackend)
+		if !ok {
+			return logical.ErrorResponse("raft storage is not in use"), logical.ErrInvalidRequest
+		}
+
+		// Get the config present in the backend
+		config := raftStorage.AutopilotConfig()
+
+		// Mutate the clone of the actual config to avoid setting values in
+		// failure cases.
+		configClone := config.Clone()
+
+		persist := false
+		cleanupDeadServers, ok := d.GetOk("cleanup_dead_servers")
+		if ok {
+			configClone.CleanupDeadServers = cleanupDeadServers.(bool)
+			persist = true
+		}
+		lastContactThreshold, ok := d.GetOk("last_contact_threshold")
+		if ok {
+			configClone.LastContactThreshold = time.Duration(lastContactThreshold.(int)) * time.Second
+			persist = true
+		}
+		maxTrailingLogs, ok := d.GetOk("max_trailing_logs")
+		if ok {
+			configClone.MaxTrailingLogs = uint64(maxTrailingLogs.(int))
+			persist = true
+		}
+		minQuorum, ok := d.GetOk("min_quorum")
+		if ok {
+			configClone.MinQuorum = uint(minQuorum.(int))
+			persist = true
+		}
+		serverStabilizationTime, ok := d.GetOk("server_stabilization_time")
+		if ok {
+			configClone.ServerStabilizationTime = time.Duration(serverStabilizationTime.(int)) * time.Second
+			persist = true
+		}
+
+		if configClone.CleanupDeadServers && configClone.MinQuorum < 3 {
+			return logical.ErrorResponse("min_quorum must be set when cleanup_dead_servers is set and it should at least be 3"), logical.ErrInvalidRequest
+		}
+
+		if persist {
+			entry, err := logical.StorageEntryJSON("core/raft/autopilot/configuration", configClone)
```

Comment:
When you do the ent PR, don't forget to add this to ignoredPrefixes so a primary's config doesn't overwrite a secondary's.

Reply:
Done